app/models/workflow.rb

```diff
@@ -29,16 +30,54 @@ def table
   end
 
   def new_sequence
-    return move_rest_higher if sequence
+    throw :abort if sequence && sequence <= 0
 
-    self.sequence = self.class.last&.sequence.to_i + 1
+    largest = last_sequence
+    if sequence && sequence < last_sequence
+      sequence_higher(sequence)
+    else
+      self.sequence = largest + 1 # auto_assignment if sequence is nil or too large
+    end
   end
 
-  # move all related sequence number one higher if the desired number is in use
-  def move_rest_higher
-    return unless sequence && sequence_changed? && self.class.exists?(:sequence => sequence)
+  # no gap between sequences
+  def adjust_sequences
+    return unless sequence_changed?
 
-    self.class.where(table[:sequence].gteq(sequence)).update_all("sequence = (-sequence - 1)")
-    self.class.where(table[:sequence].lt(0)).update_all("sequence = (-sequence)")
+    throw :abort if sequence && sequence <= 0
```

Comment:
@bzwei Should this raise an InvalidArguments Exception so the user knows that they have supplied an invalid parameter

Reply:
I learned this from rails doc. `throw :abort` will guide rails to raise an `RecordInvalid` exception.